src/HSMP/HSMPRequest.cpp

```diff
@@ -59,11 +59,13 @@ void MessageRequest::PrintStructure() const {
 char* MessageRequest::ParseToCharBuffer() const {
   std::string parsed_structure("m");
 
-  if (this->tam_msg <= 9)
-    parsed_structure += "00";
-  else if (this->tam_msg <= 99)
+  if (this->msg.length() <= 99) {
```

Comment:
Por qué usas `this->msg.length()` cuando ya tenemos el tamaño directamente en `this->tam_msg`?